eureka/S5_lightcurve_fitting/plots_s5.py

```diff
@@ -324,5 +311,87 @@ def plot_res_distr(lc, model, meta, fitter):
             plt.close()
         else:
             plt.pause(0.2)
-
     return
+        
+        
+        
+
+def plot_GP_components(lc, model, meta, fitter, isTitle=True):
+    """Plot the GP model
+    Parameters
+    ----------
+    lc: eureka.S5_lightcurve_fitting.lightcurve.LightCurve
+        The lightcurve data object
+    model: eureka.S5_lightcurve_fitting.models.CompositeModel
+        The fitted composite model
+    meta: MetaClass
+        The metadata object
+    fitter: str
+        The name of the fitter (for plot filename)
+    Returns
+    -------
+    None
+    Notes
+    -----
+    History:
+    - February 28, 2022 Eva-Maria Ahrer
+        Written function
+    - March 9, 2022 Eva-Maria Ahrer
+        Adapted with shared parameters
+    """
+    
+    if type(fitter)!=str:
+        raise ValueError('Expected type str for fitter, instead received a {}'.format(type(fitter)))
+    
+    
+    model_with_GP = model.eval(incl_GP = True)
+    model_sys_full = model.syseval()
+    model_phys_full, new_time = model.physeval(interp=meta.interp)
+    model_lc = model.eval()
+    model_GP = model.GPeval(model_lc)
+    
+    for i, channel in enumerate(lc.fitted_channels):
+        flux = np.copy(lc.flux)
+        if "unc_fit" in lc.__dict__.keys():
+            unc = deepcopy(lc.unc_fit)
+        else:
+            unc = np.copy(lc.unc)
+        model = np.copy(model_with_GP)
+        model_sys = model_sys_full
+        model_phys = model_phys_full
+        model_GP_component = model_GP
+        color = lc.colors[i]
+
+        if lc.share:
+            flux = flux[channel*len(lc.time):(channel+1)*len(lc.time)]
+            unc = unc[channel*len(lc.time):(channel+1)*len(lc.time)]
+            model = model[channel*len(lc.time):(channel+1)*len(lc.time)]
+            model_sys = model_sys[channel*len(lc.time):(channel+1)*len(lc.time)]
+            model_phys = model_phys[channel*len(new_time):(channel+1)*len(new_time)]
+            model_GP_component = model_GP_component[channel*len(lc.time):(channel+1)*len(lc.time)]
+        
+        residuals = flux - model
+        fig = plt.figure(int('51{}'.format(str(channel).zfill(len(str(lc.nchannel))))), figsize=(8, 6))
```

Comment:
Plot number 51 has already been used - we should likely use a different number here and in the plot file name

Reply:
I thought it'd be good to have the two light curve plots in packs, but can change the number of course if you think that's confusing.

Reply:
Hmm, let's see what Kevin thinks

Reply:
No to repeating 51. My suggestion would be to use 52 and relabel the current 52 to something else.

Reply:
Okay, thanks Kevin!
I've now changed the GP plot to 55 (i.e. the last number) so that it won't get confusing to change the 52 plots, especially when users don't run GPs that there aren't missing numbered plots.

Reply:
Note that you're still using plt.figure(51...) though, so that line should also be 55

Reply:
Can you change this line to the following so that only one plot gets opened (and future channels overwrite the plot) to limit RAM usage.

`fig = plt.figure(int('51{}'.format(str(0).zfill(len(str(lc.nchannel))))), figsize=(8, 6))`